### ISSUE TITLE
Link component in app directory not working properly in terms of scroll behaviour

### ISSUE THREAD

**Issue by jonasg339** [vercel/next.js#49255]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro
    Binaries:
      Node: 18.14.2
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 13.4.0
      eslint-config-next: 13.3.4
      react: 18.2.0
      react-dom: 18.2.0
```

### Which area(s) of Next.js are affected? (leave empty if unsure)

App directory (appDir: true), Routing (next/router, next/navigation, next/link)

### Link to the code that reproduces this issue

https://codesandbox.io/p/sandbox/condescending-swartz-51013l?file=%2Fapp%2Flayout.tsx&selection=%5B%7B%22endColumn%22%3A5%2C%22endLineNumber%22%3A13%2C%22startColumn%22%3A5%2C%22startLineNumber%22%3A13%7D%5D

### To Reproduce

One <Link> element from next/link references to a page with a div at the top of the page that has a position:fixed and one <Link> element references to a page without that element. 
Click either <Link> element from the initial page (scroll to the bottom and click the respective buttons) and the scroll restoration behaves totally different. 

### Describe the Bug

On the page without the position:fixed element, the clicked page starts at the top and you can scroll down normally. On the page with that fixed element, the scroll position of the previous page is taken over and the page starts at the bottom in this example. This can lead to quite confusing navigation behaviours.

(not sure if the position:fixed element is the only thing that leads to this bug)

And in general the "scroll" property of <Link> doesnt influence the behaviour at all.

Doesnt change when site is deployed

### Expected Behavior

A Link to either of the pages should lead to the pages scroll position starting at the top. The page with the fixed element shouldnt keep its scroll position from the previous page.

**Comment by wonu** [comment]
Same here, and `position: sticky` with inset properties like `top: 0` also have the same issue.

**Comment by stay-js** [comment]
Same problem here, with fixed navbar. `next: 13.4.1`

**Comment by stay-js** [comment]
It turned out that this issue only occurs when I include a link referencing an id like `/page#id` in the navbar, and even if I click the link pointing to `/page` it scrolls to `/page#id`, and on other pages it just simply doesn't scroll to the top.
If I remove the link pointing to `/page#id` the issue goes away.

The same issue have never happened using pages router.

**Comment by unleashit** [comment]
I see a number of issues relating to route changes and scroll position so I decided to add a comment on one rather than post a new one (even though my situation isn't a perfect match). I made post in the discussions section that goes into detail: https://github.com/vercel/next.js/discussions/49263.

Basically I'm having two issues: #1 links that lead to pages with nested layouts arrive at the scroll position of the top of the layout, instead of the top of the page (when there is vertical content from the parents) and #2 when you click on links lower down on a page (position sticky not necessary, any link) the router remembers and always returns to the link's position when linked to from anywhere else, In both cases, it makes no sense to arrive part way down the page.
